src/main/java/com/example/bookstore/controller/users/BookDetailController.java

```diff
@@ -0,0 +1,31 @@
+package com.example.bookstore.controller.users;
+
+import com.example.bookstore.entity.Book;
+import com.example.bookstore.entity.BookCategory;
+import com.example.bookstore.service.BooksService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Controller
+public class BookDetailController {
+
+    @Autowired
+    private BooksService booksService;
+
+    @GetMapping("/book/{id}")
+    public String getBookDetail(@PathVariable Integer id, Model model) {
```

Comment:
```suggestion
    public String show(@PathVariable Integer id, Model model) {
```
suggest nên tìm hiểu thêm về tên method, action restful cho nso chuyên nghiệp ấy